src/rdkafka_queue.c

```diff
@@ -516,8 +516,33 @@ int rd_kafka_q_serve (rd_kafka_q_t *rkq, int timeout_ms,
 	return cnt;
 }
 
+/**
+ * @brief Filter out and destroy outdated messages.
+ *
+ * @returns Returns the number of valid messages.
+ *
+ * @locality Any thread.
+ */
+static size_t rd_kafka_purge_outdated_messages (int32_t version,
+                                  rd_kafka_message_t **rkmessages,
+                                  int cnt) {
```

Comment:
Mix of size_t and int, pick one.
Looks like the caller is using unsigned int, so size_t is preferable.